packages/core/admin/admin/src/hooks/useAdminRoles.ts

```diff
@@ -29,16 +43,18 @@ export const useAdminRoles = (params = {}, queryOptions = {}) => {
   // value, which later on triggers infinite loops if used in the
   // dependency arrays of other hooks
   const roles = React.useMemo(() => {
-    let roles = [];
+    let roles: RoleEntity[] = [];
 
-    if (id && data) {
-      roles = [data.data];
-    } else if (Array.isArray(data?.data)) {
-      roles = data.data;
+    if (data) {
+      if (Array.isArray(data.data)) {
+        roles = data.data;
+      } else {
+        roles = [data.data];
+      }
     }
```

Comment:
Probably irrelevant for this case, but probably in general is better to [avoid logic refactoring for the TS migration ](https://www.notion.so/strapi/Typescript-Migration-Suggestions-and-Tips-52d61cadf4b74aa2b2b73d42de40f472?pvs=4#6820c1a353ea4586a3ae164536cfbd39)(here the id check is not present) 

Again probably doesnt matter at all here, just wanted to use this as an opportunity to share this

Reply:
aaah, classic TS